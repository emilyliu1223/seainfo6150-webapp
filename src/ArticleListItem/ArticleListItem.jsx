
import React,{useState} from 'react'
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton"
import styles from"./ArticleListItem.module.css"
const ArticleListItem=(props)=> {
    
  
    const [isTextShowing, setIsTextShowing] = useState(false);
  
    function onClick() {
      setIsTextShowing(!isTextShowing);
    }
  
    return (
      <div className={styles.container}>
      <article className={styles.article}>
        <div className={styles.wrapper}>
          <h2 className={styles.title}>{props.article.title}</h2>
          {isTextShowing && (
            <div className={styles.text}>
              <p>{props.article.shortText}</p>
              <time className={styles.time} dateTime={props.article.timeStamp}>
                {props.article.displayDate}
              </time>
            </div>
          )}
        </div>
        <div className={styles.button}> 
        <ArticleTextToggleButton
          buttonText={isTextShowing ? "Show less" : "Show more"}
          onClick={onClick}
        />
        </div>
      </article>
    </div>

    )
}


export default ArticleListItem
