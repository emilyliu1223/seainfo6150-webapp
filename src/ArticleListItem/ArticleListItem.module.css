article{
    margin: auto;
    width: 70%;
    border: pink 0.2rem solid;
    background-color: gray;
}
h1{
    color: rgb(207, 23, 23);
    margin-left: 1.3rem;
    font-weight: 800;
}
 .section{
    background-image: linear-gradient(to right,rgb(255, 0, 0,0.3) 3%, white 15%, white 30%);
   margin: 0.5rem;
   border-radius: 0.5rem;
   margin: 1.0rem;
  border: red 0.1rem solid;

}
/* .btn{
    color: red;
} */
.display{
    margin-left: 0.8rem;
}