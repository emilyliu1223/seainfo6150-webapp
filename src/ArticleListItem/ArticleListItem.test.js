import React,{useState} from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";


  



describe("ArticleListItem tests", () => {
  it("renders correctly", () => {
    const { container } = render(<ArticleListItem article={{title:"hey",image:{_url:"aaa"}, shortText:"hello",
  timeStamp:"12/12/2019",displayDate:"05/03/2020"}} key={'1'}/>);
    expect(container).toMatchSnapshot();
  });
});

