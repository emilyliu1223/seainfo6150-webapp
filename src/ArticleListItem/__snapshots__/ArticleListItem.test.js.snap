// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <div
          class="image_container"
        >
          <img
            alt="hey"
            class="image"
            src="aaa"
          />
        </div>
        <h2
          class="title"
        >
          hey
        </h2>
      </div>
      <div
        class="button"
      >
        <button
          class="button"
        >
          Show more
        </button>
      </div>
    </article>
  </div>
</div>
`;
